### PR TITLE
Enable assertions for tests

### PR DESCRIPTION
**Pull request by michalszynkiewicz**
This change enables assertions for test classloaders and passes through assertion settings for other classloaders.

### CODE REVIEW

**Comment by michalszynkiewicz** [comment]
most (all?) of the failures are because assertions in `MapBuildTimeConfigBuildStep` fail.
I think I can fix it but would like to hear from you @stuartwdouglas first if you think we want this change.

**Review by stuartwdouglas — CHANGES_REQUESTED** [review]
IMHO this should be an explicit builder field, and we set the default to the JVM default, so it can be explicitly enabled for test class loaders.

**Comment by michalszynkiewicz** [comment]
makes sense, I'll rework this next week (public holiday + PTO next two days)

**Comment by gsmet** [comment]
@michalszynkiewicz ping :)

**Comment by michalszynkiewicz** [comment]
this issue got a bit lost on my roadmap :D I'll try to get back to it t his week.

File: core/test-extension/deployment/src/test/java/io/quarkus/config/BuildTimeConfigTest.java [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+
+    @RegisterExtension
+    static final QuarkusUnitTest config = new QuarkusUnitTest()
+            .overrideConfigKey("test-map-config", testFile.getAbsolutePath());
+
+    /**
+     * The checks are done before the test is started, in {@link io.quarkus.extest.deployment.MapBuildTimeConfigBuildStep}
+     * here we only verify if the build step did its job
+     */
+    @Test
```
Comment (michalszynkiewicz) [inline core/test-extension/deployment/src/test/java/io/quarkus/config/BuildTimeConfigTest.java:39]: @radcortez this is what I ended up doing for the assertions that were failing

**Comment by michalszynkiewicz** [comment]
> IMHO this should be an explicit builder field, and we set the default to the JVM default, so it can be explicitly enabled for test class loaders.

@stuartwdouglas updated, WDYT?

**Comment by stuartwdouglas** [comment]
Failures look related, in a 'showing up existing problems way'. I will try and have a look today.

File: core/test-extension/deployment/src/test/java/io/quarkus/config/BuildTimeConfigTest.java (hunk 1 above)
Comment (radcortez) [reply core/test-extension/deployment/src/test/java/io/quarkus/config/BuildTimeConfigTest.java:39]: Sounds good.

File: independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java [hunk 2]
```diff
@@ -168,7 +170,7 @@ public CuratedApplication bootstrap() throws BootstrapException {
                 appModelFactory.setEnableClasspathCache(true);
             }
         }
-        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig);
+        return new CuratedApplication(this, appModelFactory.resolveAppModel(), classLoadingConfig, assertionsEnabled);
```
Comment (stuartwdouglas) [inline independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/app/QuarkusBootstrap.java:173]: You don't need to pass this in as a constructor argument, as it has access to QuarkusBootstrap.

**Review by stuartwdouglas — APPROVED** [review]

**Comment by stuartwdouglas** [comment]
https://github.com/quarkusio/quarkus/pull/18508

**Pull request merged**